### PR TITLE
Add internal column to ServiceTemplate table

### PR DESCRIPTION
**Pull request by hsong-rh**
Add a internal column in ServiceTemplate table to mark its accessible scope.

https://bugzilla.redhat.com/show_bug.cgi?id=1594408

### CODE REVIEW

**Comment by hsong-rh** [comment]
This is related with PR https://github.com/ManageIQ/manageiq/pull/17681

@bzwei @gmcculloug Please review.

**Comment by bzwei** [comment]
LGTM

**Comment by gmcculloug** [comment]
@miq-bot add_label transformation
@miq-bot assign @Fryguy

**Comment by hsong-rh** [comment]
`
  include ReservedMixin
`
`
  reserve_attribute :resource_id, :big_integer
` 
need to be added in ServiceTemplate to pass CI build.

**Comment by bthurber** [comment]
@gmcculloug any remaining items left to merge?

**Comment by gmcculloug** [comment]
@bthurber I do not have permission to merge on this repo.

@Fryguy Please review.

File: spec/migrations/20180719163110_add_internal_to_service_template_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+require_migration
+
+describe AddInternalToServiceTemplate do
+  let(:reserve_stub)          { Spec::Support::MigrationStubs.reserved_stub }
+  let(:service_template_stub) { migration_stub(:ServiceTemplate) }
+
+  migration_context :up do
+    it "migrate reserve data to ServiceTemplate" do
```
Comment (bzwei) [inline spec/migrations/20180719163110_add_internal_to_service_template_spec.rb]: Can you add a test case without reserved data but with `ServiceTemplateTransformationPlan` type?

**Comment by JPrause** [comment]
@miq-bot add_label blocker

**Comment by JPrause** [comment]
@hsong-rh if this can be backported, can you add the gaprindashvili/yes label.

**Comment by gmcculloug** [comment]
@JPrause We do not back-port schema changes.

The work for the BZ in the description, that will be back-ported, is in https://github.com/ManageIQ/manageiq/pull/17748 and is already properly labeled.  This PR is related to that work but is intended for upstream only.

**Comment by gmcculloug** [comment]
@Fryguy Please review

**Pull request merged**